characterEntities/Entity.java

```diff
@@ -203,6 +207,9 @@ public void setFacingEast(boolean facingEast) {
 	}
 
 	public void setAnimation(int index, AbilityAnimation abilityAnimation) {
+		if (abilityAnimations.size() > index) {
```

Comment:
Since we populated the array with nulls, each null actually counts as an object, so the list.size() is always 5. Therefore, this actually makes it consistently of size 5